hail/src/main/scala/is/hail/types/physical/stypes/interfaces/SStruct.scala

```diff
@@ -23,9 +24,15 @@ trait SBaseStructValue extends SValue {
   def loadField(cb: EmitCodeBuilder, fieldName: String): IEmitSCode = loadField(cb, pt.fieldIdx(fieldName))
 }
 
-trait SBaseStructCode extends SCode {
+trait SBaseStructCode extends SCode { self =>
+  def st: SStruct
```

Comment:
this isn't valid -- these can represent tuples too.

Reply:
There's no issue here. There is no STuple, the code need only be a base struct.

Reply:
ah, right. Carry on.